src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,260 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(null);
+        },
+                "It must be thrown UserRegistrationException for null user");
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setPassword("thisPasswordIsFit");
+        user.setAge(21);
+        user.setLogin(null);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },
+                "It must be thrown UserRegistrationException for null login");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        User user = new User();
+        user.setPassword("thisPasswordIsFit");
+        user.setAge(21);
+        user.setLogin("");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },
+                "It must be thrown UserRegistrationException for empty login");
+    }
+
+    @Test
+    void register_loginIsLessThanMinLength_notOk() {
+        User user = new User();
+        user.setPassword("thisPasswordIsFit");
+        user.setAge(21);
+        user.setLogin("12345");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(user);
+        },
+                "It must be thrown UserRegistrationException for login"
+                        + user.getLogin());
+    }
+
+    @Test
+    void register_loginMinLength_Ok() {
+        User expected = new User();
+        expected.setLogin("login6");
+        expected.setPassword("thisPasswordIsFit");
+        expected.setAge(21);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_loginIsFit_Ok() {
+        User expected = new User();
+        expected.setPassword("thisPasswordIsFit");
+        expected.setAge(21);
+        expected.setLogin("login@1454sd");
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
+    }
```

Comment:
basically one _ok test is enough for this class. Pass the user which has all valid fields, make necessary assertions